tests/test_fmi3.py

```diff
@@ -390,6 +465,36 @@ def test_set_array_variable(self):
         with pytest.raises(FMUException, match = err_msg):
             fmu.set("x", np.array([1, 2, 3]))
 
+    def test_get_continuous_states(self):
+        """Test retrieve the continuous states."""
+        fmu_path = FMI3_REF_FMU_PATH / "VanDerPol.fmu"
+        fmu = load_fmu(fmu_path)
+        fmu.initialize()
+        assert all(fmu.continuous_states == [2.0, 0.0])
+
+    @pytest.mark.parametrize("variable_name, expected_datatype",
+        [
+            ("Float64_continuous_input", FMI3_Type.FLOAT64),
+            ("Float32_continuous_input", FMI3_Type.FLOAT32),
+            ("Int64_input", FMI3_Type.INT64),
+            ("Int32_input", FMI3_Type.INT32),
+            ("Int16_input", FMI3_Type.INT16),
+            ("Int8_input" , FMI3_Type.INT8),
+            ("UInt64_input", FMI3_Type.UINT64),
+            ("UInt32_input", FMI3_Type.UINT32),
+            ("UInt16_input", FMI3_Type.UINT16),
+            ("UInt8_input",  FMI3_Type.UINT8),
+            ("Boolean_input", FMI3_Type.BOOL),
+            ("String_parameter", FMI3_Type.STRING),
+            ("Binary_input", FMI3_Type.BINARY),
+            ("Enumeration_input", FMI3_Type.ENUM),
+        ]
+    )
+    def test_get_variable_data_type(self, variable_name, expected_datatype):
+        """Test getting variable data types."""
+        fmu_path = FMI3_REF_FMU_PATH / "Feedthrough.fmu"
+        fmu = FMUModelME3(fmu_path, _connect_dll = False)
+        assert fmu.get_variable_data_type(variable_name) is expected_datatype
 
```

Comment:
Maybe I missed it, but some suggestions for 2 extra tests:

```
states = ...
set(states)
assert states == get(states)
```

```
ders_1 = get_derivatives(...)
states = ...
set(states)
ders_2 = get_derivatives(...)
assert ders_1 != ders_2 
```

Or similar.

Reply:
Good idea, see the new tests:
def test_get_and_set_states(self)
def test_get_derivatives_with_states_set(self)